### DISCUSSION TITLE
Avoid Pod problems by unavailable CIFS Share

### DISCUSSION THREAD

**Discussion by donkon** [kubernetes-csi/csi-driver-smb#845] [General] [answered]
We mount multiple cifs volumes by the csi-driver-smb driver into one pod.

If one CIFS share goes down (and the others are still up and running), the pod goes down.

Is there a way to come around this? We would like to have our pod up and running, even if one share is not available for some time.

**Comment by andyzhangx** [comment]
that depends on your application logic, you need to change your application code to make sure your application could still live if one volume path is unavailable

**Reply by donkon** [reply]
The pod goes down, so app logic can not help

**Reply by andyzhangx** [reply, marked as answer]
the pod goes down because your application continuously writes data to the volume, and if there is write error, your application would crash, so if your pod could just raise error instead of crash in that case, that should help.

**Reply by donkon** [reply]
you are right, we have a liveness probe checking the volumes, this forced the pod to restart which did not succeed because of the not working cifs share - I changed the application logic (LivenessProbe) and works now

**Reply by andyzhangx** [reply]
@donkon thanks for the update, could you paste config about `liveness probe checking the volumes`? this may be helpful for others, thx

**Reply by donkon** [reply]
Well, our custom HealthIndicator (written in Spring Boot) endpoint behind 
```
          livenessProbe:
            httpGet:
              path: /wfs/actuator/health
              port: 8080
            initialDelaySeconds: 30
            periodSeconds: 60
```
did things wrong (tried to access the not working samba server), so I did not change the config - just the implementation of a our custom code (to not access the mounted filesystem any more)
